vermouth/ffinput.py

```diff
@@ -935,14 +935,12 @@ def _parse_edges(tokens, context, context_type, negate):
         raise IOError('The "non-edges" section is only valid in links.')
     atoms = _get_atoms(tokens, natoms=2)
     prefixed_atoms = []
-    for atom in atoms:
+
+    for idx, atom in enumerate(atoms):
         prefixed_reference, attributes = _treat_atom_prefix(*atom)
-        try:
-            apply_to_all_nodes = context._apply_to_all_nodes
-        except AttributeError:
-            apply_to_all_nodes = {}
-        full_attributes = dict(collections.ChainMap(attributes, apply_to_all_nodes))
+        full_attributes = dict(collections.ChainMap(attributes, {}))
         prefixed_atoms.append([prefixed_reference, full_attributes])
+
```

Comment:
Why is this change in this PR?

Reply:
some cleaning up; might as well do it here.

Reply:
How do you deal with the attribute to apply to all nodes, then? I do not see the feature used, though.

If you want to remove this, why keep the `ChainMap`?

Reply:
@jbarnoud Can you explain what the purpose of the "_apply_to_all_nodes" attribute is? I've only removed the except statement because only the Link class has a _apply_to_all_nodes attribute and that just results in an empty dict. I thought this was pointless as that is the same as just giving it an empty dict. However, I am happy to be convinced otherwise.

Reply:
Ok, got to dig back deeper in my memory. It may actually be used, even. Look at this example:

```
[ link ]
resname $protein_resnames
cgsecstruct "F"
[ dihedrals ]
-BB BB  +BB ++BB 1 90.7 100 1
[ features ]
collagen
```

This is equivalent to:

```
[ link ]
[ dihedrals ]
-BB {"resname": $protein_resnames, "cgsecstruct": "F"} BB {"resname": $protein_resnames, "cgsecstruct": "F"} +BB {"resname": $protein_resnames, "cgsecstruct": "F"} ++BB {"resname": $protein_resnames, "cgsecstruct": "F"} 90.7 100 1
[ features ]
collagen
```

Nobody wants to type that. Here, the attributes just after `[ link ]` go to the `_apply_to_all_nodes` attribute and is applied to all the nodes described in the subsequent sections, including `[ edges ]`.

Reply:
I added it back. No harm done